packages/bot-web-ui/src/pages/bot-builder/quick-strategy/form-wrappers/upgraded-qs-v2/strategy-list.tsx

```diff
@@ -0,0 +1,132 @@
+                            <div key={index} className='strategy-template-picker__strategy'>
+                                {has_visible_strategies && (
+                                    <div className='strategy-template-picker__title'>
+                                        <Text size='xs' weight='bold'>
+                                            {trade_type}
+                                        </Text>
+                                    </div>
+                                )}
+                                {strategies.map(
+                                    (strategy: TStrategyValue, idx: number) =>
+                                        shouldShowStrategies(trade_type, strategy.label) && (
+                                            <div className='strategy-template-picker__links' key={idx}>
+                                                <Link
+                                                    color={'var(--text-prominent)'}
+                                                    hasChevron
+                                                    size='sm'
+                                                    onClick={() => onSelectStrategy(strategy.name, trade_type)}
+                                                >
+                                                    {strategy.label}
+                                                </Link>
+                                            </div>
+                                        )
+                                )}
+                            </div>
+                        )
+                    );
+                })
+            ) : (
+                <div className='strategy-template-picker__strategy'>
+                    {is_content_visible && (
+                        <div className='strategy-template-picker__title'>
+                            <Text size='xs' weight='bold'>
+                                {trade_types[option]}
+                            </Text>
+                        </div>
+                    )}
+                    {trade_type_value.map(
+                        (strategy: TStrategyValue, index: number) =>
+                            shouldShowStrategies(trade_types[option], strategy.label) && (
+                                <div className='strategy-template-picker__links' key={index}>
+                                    <Link
+                                        color={'var(--text-prominent)'}
+                                        hasChevron
+                                        size='sm'
+                                        onClick={() => onSelectStrategy(strategy.name, trade_types[option])}
+                                    >
+                                        {strategy.label}
+                                    </Link>
+                                </div>
+                            )
+                    )}
+                </div>
+            )}
+        </div>
+    );
```

Comment:
@maryia-matskevich-deriv there is a lot going on here. It is hard to read. Can you explain what is the purpose of this function and also can we simplify it to improve the readability ?

Reply:
logic changed